libr/bin/p/bin_elf.inc

```diff
@@ -715,6 +715,17 @@ static RBinReloc *reloc_convert(struct Elf_(r_bin_elf_obj_t) *bin, RBinElfReloc
 		break;
 		}
 		break;
+	case EM_BPF: switch (rel->type) {
+		case R_BPF_NONE: break;
+		case R_BPF_64_64: r->vaddr += 4; ADD (32, 0);
```

Comment:
I have no idea whether I've done this correctly.
According to Linux docs:

> For example, R_BPF_64_64 relocation type is used for ld_imm64 instruction. The actual to-be-relocated data (0 or section offset) is stored at r_offset + 4 and the read/write data bitsize is 32 (4 bytes). The relocation can be resolved with the symbol value plus implicit addend. Note that the BitSize is 32 which means the section offset must be less than or equal to UINT32_MAX and this is enforced by LLVM BPF backend.

Again, the `ld_imm64` instruction uses split encoding (4x32-bit words, high 32-bits in word 1, low 32-bits in word 3). So the value to be written gets forcibly truncated to 32 bits. I think the Linux kernel does this because it has a 32-bit program counter, but this doesn't make sense for a 64-bit PC flavor.

Reply:
I would say that if the resulting pointers look good it's fine to merge and start adding tests with more binaries.